cfme/tests/automate/test_vmware_methods.py

```diff
@@ -23,7 +23,7 @@
 
 
 def pytest_generate_tests(metafunc):
-    argnames, argvalues, idlist = testgen.provider_by_type(
+    argnames, argvalues, idlist = testgen.providers_by_type(
```

Comment:
See I totally understand your point about the strings not importing anything, but the objects would. I wonder if we can create a module like the test_requirements one, that maps the strings to attribute names, so we get the benefit of one without the other. Initially these would be stored in a single file which would increase the places that we need to modify when adding new providers, BUT I think ultimately each provider would become a package and would have an init that provided the attribute and was handled and registered at import time and then another module that held the REAL class, so we get hte best of both, but that's a little way down the line....thoughts??